### DISCUSSION TITLE
Can I generate my own cuid?

### DISCUSSION THREAD

**Discussion by odm275** [prisma/prisma#15145] [Q&A] [answered]
My question stems from me not knowing enough about cuid, but would there be a problem if I created my own cuid ids when creating an Item in the database? Like with a library like https://github.com/ericelliott/cuid ?

**Comment by ludralph** [comment, marked as answer]
Hi @odm275 👋 

There won't be issues if you create your own cuid ids for a record in the database.

An example of a sample schema is 
```js
model User {
  id  String  @id @unique
  lastname String
}
```
Then from the script.js file you can do something like 
```js
const cuid = require('cuid');
const user = await prisma.user.create({
  data: {
    id: cuid(),
    lastname: 'test lastname',
  }
})
console.log(user)
```
The output is 
```
{ id: 'cl7kfwarz000036vgdkyqdwzf', lastname: 'test lastname' }
```
Also you can leverage the builtin `cuid` in the prisma schema as well. You would structure your schema like the example below
```js
model User {
  id  String  @id @unique @default(cuid())
  lastname String
}
```
then within the script file, tou would do something like 
```
const user = await prisma.user.create({
  data: {
    lastname: 'test lastname',
  }
})
console.log(user)
```
The output will be
```
{ id: 'cl7kg4rpz000051vgkwhle9rv', lastname: 'test lastname' }
```

**Reply by odm275** [reply]
Thanks for the response @ludralph!  I understand it works on paper, but I was wondering if it there would be any reason why it wouldn't in practice. I'm just very ignorant about how cuid generation works, and maybe prisma has its own special way of generating cuids that is counter to how some other library would generate them, and maybe there is some interference there was my thought.  But it seems like it is all good!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
